runtime/src/bank.rs

```diff
@@ -7775,17 +7917,43 @@ pub(crate) mod tests {
                     slot: expected_previous_slot,
                     ..Clock::default()
                 },
-                1,
+                bank1.inherit_specially_retained_account_balance(optional_account),
             )
         });
         let current_account = bank1.get_account(&dummy_clock_id).unwrap();
         assert_eq!(
             expected_previous_slot,
             from_account::<Clock>(&current_account).unwrap().slot
         );
+        let new_capitalization = bank1.capitalization();
+        // only if simpler_capitalization_enabled, cap should increment
+        if simpler_capitalization_enabled {
+            assert_eq!(old_capitalization + 1, new_capitalization);
+        } else {
+            assert_eq!(old_capitalization, new_capitalization);
+        }
+        assert_eq!(bank1.calculate_capitalization(), bank1.capitalization());
+
+        let old_capitalization = bank1.capitalization();
+        bank1.update_sysvar_account(&dummy_clock_id, |optional_account| {
+            assert!(optional_account.is_none());
+
+            create_account(
+                &Clock {
+                    slot: expected_previous_slot,
+                    ..Clock::default()
+                },
+                bank1.inherit_specially_retained_account_balance(optional_account),
+            )
+        });
+        let new_capitalization = bank1.capitalization();
+        // creating new sysvar twice in a slot shouldn't increment capitalization twice
+        assert_eq!(old_capitalization, new_capitalization);
```

Comment:
Very subtle bug was caught here.

Combined with new use of `inherit_specially_retained_account_balance` in this (now-parameterized) test, this new code blocks illustrates why it was a bad idea to update bank.capitalization in `inherit_specially_retained_account_balance` (this was my original draft pr changes)...